components/TutorialCard.js

```diff
@@ -1,8 +1,40 @@
 import Image from "next/image";
+import { useEffect, useState } from "react";
 import styled from "styled-components";
 import { Paragraph, Subline } from "./TextElements";
 
-export default function TutorialCard({ step }) {
+export default function TutorialCard({ step, totalSteps, setCurrentStep }) {
+  const [stepNumberInput, setStepNumberInput] = useState(step.step);
+
+  function handleStepChange(event) {
+    const eventValue = event.target.value;
+    const sanitizedEventValue = eventValue.trim().replace(".", "");
+    if (
+      sanitizedEventValue === "" ||
+      (sanitizedEventValue > 0 && sanitizedEventValue <= totalSteps)
+    ) {
+      setStepNumberInput(sanitizedEventValue);
+    }
+  }
+
+  function handleKeyDown(event) {
+    if (event.key === "Enter") {
+      jumpToStep();
+    }
+  }
+
+  function jumpToStep() {
+    if (stepNumberInput !== "") {
+      setCurrentStep(stepNumberInput);
+    } else {
+      setStepNumberInput(step.step);
+    }
+  }
+
+  useEffect(() => {
+    setStepNumberInput(step.step);
```

Comment:
I m not sure if it s necessary to set the step here. As I understand, every time the step changes, a tutorial card is called, hence the initial set at line 7 should be enough

Reply:
You are right, @anhthu253 ! I don't need the useEffect here anymore.

Reply:
I found out what I needed the useEffect for: If I don't enter the step number in the input field but use the prev or next buttons instead, the step number wouldn't update without the useEffect. So it's back in the code. 😁

Reply:
that what i assumed, that the step value in props is changed when next or prev buttons is clicked. But any action on these button also leads to a tutorialcard being invoked, isnt it? If it s so then the state step number (stepNumberInput ) is also initialized according to the step value in props. That s why i did not understand why stepNumberInput has to be set again inside useEffect
But yeah i can be wrong ;)